src/runtimes/node/utils/zip.ts

```diff
@@ -229,6 +238,58 @@ const addStat = async function (cache: RuntimeCache, srcFile: string) {
   return { srcFile, stat }
 }
 
+const ensurePackageJSONWithModuleType = async function ({
+  archive,
+  basePath,
+  mainFile,
+  name,
+  userNamespace,
+}: {
+  archive: ZipArchive
+  basePath: string
+  mainFile: string
+  name: string
+  userNamespace: string
+}) {
+  const functionDirectory = dirname(mainFile)
+
+  let packageJSON: PackageJsonFile | null = null
+
+  try {
+    packageJSON = await getClosestPackageJson(functionDirectory)
```

Comment:
do we stop at the repository root/site root here?

Reply:
Good question. I don't think that `get-tsconfig` has any option to define boundaries, so I think we'll need to look at the path it finds and assert whether it's inside or outside the repository root. I'll do that in a bit.

Reply:
Done in https://github.com/netlify/zip-it-and-ship-it/pull/1505/commits/ba79d06bbe0ab7b4e167ca621ab2645e0abe6bbb.

Reply:
Wait, I've done this for the `tsconfig.json`, but your comment is about `package.json`. Let me to do that here too.

Reply:
Done that in https://github.com/netlify/zip-it-and-ship-it/pull/1505/commits/71bb5eee988c6a8199947441a5a49b3dd9b02e60.